File: 1-two-sum/1-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int res[] = new int[2];
        HashMap<Integer,Integer> hm = new HashMap<>();
        int n = nums.length;
        for(int i=0;i<n;i++)
        {
            if(hm.containsKey(target-nums[i]))
            {
                res[0] = i;
                res[1] = hm.get(target-nums[i]);
                return res;
            }
            hm.put(nums[i],i);
        }
        return res;
    }
}

File: 1-two-sum/1-two-sum.py
class Solution {
    public int[] twoSum(int[] arr, int target) {
        int res[] = new int[2];
        int n = arr.length;
        Map<Integer,Integer> hm = new HashMap<Integer, Integer>();
        for(int i=0;i<n;i++)
        {
            if(hm.containsKey(target-arr[i]))
            {
                res[0] = i;
                res[1] = hm.get(target-arr[i]);
                return res;
            }
            else
            {
                hm.put(arr[i], i);
            }
        }
        return res;
    }
}